### PR TITLE
Fix final metrics collection when metrics data is not at beginning of line

### PR DESCRIPTION
**Pull request by chicm-ms**
Fix issue #1207

### CODE REVIEW

File: test/metrics_test/trial.py [hunk 1]
```diff
@@ -2,9 +2,14 @@
 import nni
 
 if __name__ == '__main__':
-    hyper_params = nni.get_next_parameter()
-
+    nni.get_next_parameter()
     for i in range(10):
+        if i % 2 == 0:
```
Comment (xuehui1991) [inline test/metrics_test/trial.py]: why we add these code?

File: tools/nni_trial_tool/log_utils.py [hunk 2]
```diff
@@ -134,7 +134,7 @@ def __init__(self, logger, log_collection, log_level=logging.INFO):
         self._is_read_completed = False
         self.process_exit = False
         self.log_collection = log_collection
-        self.log_pattern = re.compile(r'^NNISDK_MEb\'.*\'$')
+        self.log_pattern = re.compile(r'NNISDK_MEb\'.*\'$')
```
Comment (QuanluZhang) [inline tools/nni_trial_tool/log_utils.py]: I suggest to keep '^', it makes the matching more restrict.

File: test/metrics_test/trial.py (hunk 1 above)
Comment (chicm-ms) [reply test/metrics_test/trial.py]: Except local mode, for other training services, metrics are printed into stdout log file with special MARK 'NNISDK_ME'.
the purpose of this code is to test whether following 2 cases can be handled correctly:
1.  Intermediate result is not at the begining of a line, such as:
`report intermediate result without end of line.NNISDK_MExxx`
2.  Intermediate result is  at the begining of a line, such as:
```
report intermediate result.
NNISDK_MExxx
```

File: tools/nni_trial_tool/log_utils.py [hunk 3]
```diff
@@ -172,11 +172,19 @@ def run(self):
...
+            metrics = None
+            if search_result:
+                collect_line = line[0:search_result.span()[0]] + line[search_result.span()[1]:]
+                metrics = search_result.group(0)
+            else:
+                collect_line = line
+
+            if self.log_collection != 'none':
+                self.queue.put(collect_line)
+            elif metrics is not None:
```
Comment (QuanluZhang) [inline tools/nni_trial_tool/log_utils.py]: elif -> if?

Comment (chicm-ms) [reply tools/nni_trial_tool/log_utils.py]: Thanks for good catch, there was a bug.  Fixed now.

File: tools/nni_trial_tool/log_utils.py (hunk 2 above)
Comment (yds05) [reply tools/nni_trial_tool/log_utils.py]: The issue is caused by ^, since that case doesn't have \r\n

**Pull request merged**